component/background/StorageService.tsx

```diff
@@ -0,0 +1,92 @@
+import {STORAGE_KEY_ID} from "../client/Constants";
+
+class StorageService {
+  private storage: chrome.storage.StorageArea;
+
+  constructor() {
+    this.storage = chrome.storage.sync;
+    this.storage.get(STORAGE_KEY_ID, (id) => {
+      // If user id is not locally stored, generate a new one.
+      if (!id) this.storage.set({STORAGE_KEY_ID: this.generateUUID()});
+    });
+  }
+
+  /**
+   * Clear the storage space and load with default settings
+   * @return {Promise<{}>} A promise whose resolve takes no argument, which runs after defaults are set
+   */
+  public reset(): Promise<{}> {
+    return new Promise((resolve) => {
+      this.storage.get(STORAGE_KEY_ID, (data) => {
+        this.storage.clear(() => {
+          this.storage.set(data, resolve);
+        });
+      });
+    });
+  }
+
+  /**
+   * Retrieves the stored value for the specified key
+   * @param key The key to look up
+   * @return {Promise<String>} The value associated with the key; undefined if not found
+   */
+  public get(key: string): Promise<string> {
+    return new Promise((resolve) => {
+      this.storage.get(key, (item) => {
+        resolve(item[key]);
+      });
+    });
+  }
+
+  /**
+   * Sets the key-value in the storage
+   * @param key The key
+   * @param value The value
+   * @return {Promise<{}>} A promise whose resolve takes no argument, which runs after the value is set
+   */
+  public set(key: string, value: string): Promise<{}> {
+    const data = {};
+    data[key] = value;
+
+    return new Promise((resolve) => {
+      this.storage.set(data, resolve);
+    });
+  }
+
+  /**
+   * Removes the key (and its associated value) from the storage
+   * @param key The key to remove
+   * @return {Promise<{}>} A promise whose resolve takes no argument, which runs after the key is removed
+   */
+  public remove(key): Promise<{}> {
```

Comment:
I see most places use `Promise<any>`
Sorry, apparently, the brackets are ignored by markdown.